OpenRA.Game/Game.cs

```diff
@@ -172,6 +172,35 @@ internal static void StartGame(string mapUID, WorldType type)
 			GC.Collect();
 		}
 
+		public static void RestartGame()
+		{
+			if (OrderManager.Connection is ReplayConnection)
```

Comment:
The disconnect special casing doesn't sit well with me - we don't do this in the normal abort case, so it shouldn't be needed here.  How about something like:

```csharp
var replay = OrderManager.Connection as ReplayConnection;
var replayName = replay != null ? replay.Filename : null;

// Disconnect from the current game
Disconnect();

// Restart the game with the same replay/mission
if (replay != null)
	JoinReplay(i.Filename);
else
{
	// the rest of the code
}
```